src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/CoreApi/FileSystemApiTest.cs

```diff
@@ -0,0 +1,1013 @@
+        }
+
+        [Fact]
+        public async Task StreamBehaviour()
+        {
+            var options = new AddFileOptions
+            {
+                ChunkSize = 3,
+                Pin = true,
+            };
+            var node = await ipfs.UnixFsApi.AddTextAsync("hello world", options);
+            var stream = await ipfs.UnixFsApi.ReadFileAsync(node.Id);
+            Assert.Equal(11, stream.Length);
+            Assert.True(stream.CanRead);
+            Assert.False(stream.CanWrite);
+            Assert.True(stream.CanSeek);
+        }
+
+        [Fact]
+        public async Task Add_HashAlgorithm()
+        {
+            var options = new AddFileOptions
+            {
+                Hash = "blake2b-256",
+                RawLeaves = true
+            };
+            var node = await ipfs.UnixFsApi.AddTextAsync("hello world", options);
+            Assert.Equal("bafk2bzaceaswza5ss4iu2ia3galz6pyo6dfm5f4dmiw2lf2de22dmf4k533ba", node.Id);
+
+            var text = await ipfs.UnixFsApi.ReadAllTextAsync(node.Id);
+            Assert.Equal("hello world", text);
+        }
+
+        [Fact]
+        public void AddFile()
+        {
+            var path = Path.GetTempFileName();
+            File.WriteAllText(path, "hello world");
+            try
+            {
+                var node = (UnixFsNode) ipfs.UnixFsApi.AddFileAsync(path).Result;
+                Assert.Equal("Qmf412jQZiuVUtdgnB36FXFX7xg5V6KEbSJ4dpQuhkLyfD", node.Id);
+                Assert.Equal(0, node.Links.Count());
+                Assert.Equal(Path.GetFileName(path), node.Name);
+            }
+            finally
+            {
+                File.Delete(path);
+            }
+        }
+
+        [Fact]
+        public void AddFile_CidEncoding()
+        {
+            var path = Path.GetTempFileName();
+            File.WriteAllText(path, "hello world");
+            try
+            {
+                var options = new AddFileOptions
+                {
+                    Encoding = "base32"
+                };
+                var node = ipfs.UnixFsApi.AddFileAsync(path, options).Result;
+                Assert.Equal("base32", node.Id.Encoding);
+                Assert.Equal(1, node.Id.Version);
+                Assert.Equal(0, node.Links.Count());
+
+                var text = ipfs.UnixFsApi.ReadAllTextAsync(node.Id).Result;
+                Assert.Equal("hello world", text);
+            }
+            finally
+            {
+                File.Delete(path);
+            }
+        }
+
+        [Fact]
+        public void AddFile_Large()
+        {
+            AddFile(); // warm up
+
+            var path = "star_trails.mp4";
+            var stopWatch = new Stopwatch();
+            stopWatch.Start();
+            var node = ipfs.UnixFsApi.AddFileAsync(path).Result;
+            stopWatch.Stop();
+            
+            // _testOutputHelper.WriteLine("Add file took {0} seconds.", stopWatch.Elapsed.TotalSeconds);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)